system/lib/pthread/library_pthread.c

```diff
@@ -418,7 +418,16 @@ EMSCRIPTEN_RESULT emscripten_wait_for_call_v(em_queued_call* call, double timeou
     double waitEndTime = now + timeoutMSecs;
     emscripten_set_current_thread_status(EM_THREAD_STATUS_WAITPROXY);
     while (!done && now < waitEndTime) {
-      r = emscripten_futex_wait(&call->operationDone, 0, waitEndTime - now);
+      // Wait at most one second in each loop iteration. That keeps CPU load low
+      // while also avoids a possible race condition, as call->operationDone
+      // can be set *after* we checked it but *before* we do the futex_wait. The
+      // futex_wait is not aware of call->operationDone which means if we wait
```

Comment:
I'm confused. Isn't the whole point of `futex_wait` to do an atomic compare-and-block operation? Why wouldn't it be aware of `call->operationDone`?

Reply:
Ah, I think you're right, while this fixes the issue the diagnosis does look wrong.